crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -81,7 +81,8 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Pa
 
         helpers::authenticate_client_secret(
             request.client_secret.as_ref(),
-            payment_intent.client_secret.as_ref(),
+            &payment_intent,
```

Comment:
`authenticate_client_secret` is called in payments_operation_core and `get_trackers` . Not necessary to call in both the places I guess.